R/munge_unaids.R

```diff
@@ -0,0 +1,181 @@
+#' @title Cleaning UNAIDS Data
+#'
+#' @description This function cleans and fetches UNAIDS Estimates/Test and Treat Data
+#' @param return_type returns either HIV Estimates or Test & Treat Data
+#' @return
+#' @export
+#'
+#' @examples
+#'  \dontrun{
+#'    munge_unaids(sheet_id = googledrive::as_id('1tkwP532mPL_yy7hJuHNAHaZ1_K_wd7zo_8AjeOe7fRs'), tab = 3, skipnum = 4)
+#' }
+#'
+munge_unaids <- function(return_type) {
+
+  #to specify NA's when reading in data
+  missing <- c("...", " ")
+
+  # Get valid pepfar list
+  pepfar_cntry <- glamr::get_outable() %>%
+    dplyr::pull(countryname)
+
+  # Google Sheet ID to original
+  sheet_id <- googledrive::as_id("1tkwP532mPL_yy7hJuHNAHaZ1_K_wd7zo_8AjeOe7fRs")
+
+  sheet_id_names <- "1vaeac7hb7Jb6RSaMcxLXCeTyim3mtTcy-a1DQ6JooCw"
+
+  #HIV estimates tab - specific
+  if (return_type == "HIV Estimates") {
+
+    gdrive_df <- suppressMessages(
+      googlesheets4::read_sheet(sheet_id, sheet = 1, skip = 5, na = missing) %>%
+        dplyr::rename(year = !!names(.[1]),
+                      iso =  !!names(.[2]),
+                      country =  !!names(.[3]))
+    )
+
+    hiv_est_names <- suppressMessages(
+      googlesheets4::read_sheet(sheet_id, range = "HIV estimates - by Year!A5:AY5") %>%
+        dplyr::rename(year = !!names(.[1]),
+                      iso =  !!names(.[2]),
+                      country =  !!names(.[3]))
+    )
```

Comment:
yay for more functions! these are the same as are ones further down, which means we can refine this into a more generic function.

```
read_rename <- function(sheetname){
skip_n <- ifelse(sheetname == "HIV estimates - by Year", 5, 4)
suppressMessages(
      googlesheets4::read_sheet(sheet_id, sheet = sheetname, skip = skip_n , na = missing) %>%
        dplyr::rename(year = !!names(.[1]),
                      iso =  !!names(.[2]),
                      country =  !!names(.[3]))
    )
}
```